lib/helpers/preserve_data.rb

```diff
@@ -0,0 +1,11 @@
+def save_data(data, file)
+  json = JSON.generate(data)
+  File.write("./lib/preserve_data/#{file}.json", json)
+end
```

Comment:
This is what I was talking about before, how is this function working?
How does it get the json from data?
I think I don't get it, kindly explain it to me so I can understand it so we all can use the same function to save the data in files

Reply:
@Trictonicmp In this function I take an array from the project, example @books_list,  and the file name, in this case, I would use 'books', then it generates a JSON string of the array example: 
- "[1,2,3,4]", 

In the case of our classes with the to_json and json_create method, it first changes the instance of the class to an object and saves it in the file name created example:
-  {"json_class":"MusicAlbum","publish_date":"2020-06-06","on_spotify":true,"id":569,"archived":false,"genre":null}

But I actually use this function to save the complete array that stores all the elements created, in this case of music albums example:
- [{"json_class":"MusicAlbum","publish_date":"2020-06-06","on_spotify":true,"id":611,"archived":false,"genre":null},{"json_class":"MusicAlbum","publish_date":"2022-05-05","on_spotify":false,"id":172,"archived":false,"genre":null},{"json_class":"MusicAlbum","publish_date":"2021-12-12","on_spotify":false,"id":388,"archived":false,"genre":null}]

And at the end line, it just creates the file in the ./lib/preserve_data/ folder which I create just to have a place to store files with the file name and ending .json. 

This method override all that is in the file before, to be sure is not repeating data, so that's why is important to recover data first in the local array and after push, a new element save the entire local array again